rpm_spec/core-agent.spec.in

```diff
@@ -564,6 +638,105 @@ fi
 %systemd_preun qubes-network.service
 %systemd_preun qubes-updates-proxy.service
 
+%if %{with selinux}
+%post selinux
+
+%selinux_modules_install %{_datadir}/selinux/packages/qubes-qfile-unpacker.pp %{_datadir}/selinux/packages/qubes-xendriverdomain.pp %{_datadir}/selinux/packages/qubes-misc.pp
+{
+flock --no-fork -- 9
+flock --no-fork -- 8
+for i in /etc/selinux/*/contexts/files/file_contexts.subs; do
+if [ -s "$i" ]; then
+contexts='### QUBES START ###\
+/rw/home /home\
+/rw/config/NM-system-connections /etc/NetworkManager/system-connections\
+/rw/bind-dirs /\
+/rw/usrlocal /usr/local\
+/rw/usrlocal/etc /etc\
+/rw/usrlocal/share /usr/share\
+/rw/usrlocal/libexec /usr/libexec\
+/rw/usrlocal/games /usr/games\
+/rw/usrlocal/bin /usr/bin\
+/rw/usrlocal/lib /usr/lib\
+/rw/usrlocal/src /usr/src\
+/rw/usrlocal/include /usr/include\
+/rw/usrlocal/sbin /usr/sbin\
+/usr/local/etc /etc\
+/usr/local/share /usr/share\
+/usr/local/libexec /usr/libexec\
+/usr/local/games /usr/games\
+/usr/local/bin /usr/bin\
+/usr/local/lib /usr/lib\
+/usr/local/src /usr/src\
+/usr/local/include /usr/include\
+/usr/local/sbin /usr/sbin\
+### QUBES END ###'
+
+sed -Ei -- "/^### QUBES START ###\$/,/^### QUBES END ###\$/{
+\$c\
+$contexts
+d
+}
+\$a\
+$contexts
+" "$i"
+else
+printf >> "$i" '### QUBES START ###
+/rw/home /home
+/rw/config/NM-system-connections /etc/NetworkManager/system-connections
+/rw/bind-dirs /
+/rw/usrlocal /usr/local
+/rw/usrlocal/etc /etc
+/rw/usrlocal/share /usr/share
+/rw/usrlocal/libexec /usr/libexec
+/rw/usrlocal/games /usr/games
+/rw/usrlocal/bin /usr/bin
+/rw/usrlocal/lib /usr/lib
+/rw/usrlocal/src /usr/src
+/rw/usrlocal/include /usr/include
+/rw/usrlocal/sbin /usr/sbin
+/usr/local/etc /etc
+/usr/local/share /usr/share
+/usr/local/libexec /usr/libexec
+/usr/local/games /usr/games
+/usr/local/bin /usr/bin
+/usr/local/lib /usr/lib
+/usr/local/src /usr/src
+/usr/local/include /usr/include
+/usr/local/sbin /usr/sbin
+### QUBES END ###
+'
+fi
+done
+} 9< %_selinux_store_policy_path/semanage.read.LOCK 8< %_selinux_store_policy_path/semanage.trans.LOCK
+if [ ! -f /.qubes-relabeled ]; then
+    unshare --mount -- /bin/sh -euc 'mount --bind -- / /mnt && exec "$@"' sh setfiles -r /mnt -- \
+      /etc/selinux/targeted/contexts/files/file_contexts \
+      /mnt/proc /mnt/sys /mnt/dev /mnt/tmp /mnt/rw /mnt/home \
+      /mnt/usr/lib/systemd
```

Comment:
Some of it may not be necessary, I see systemd prints very early this:
```
[    8.965076] systemd[1]: Relabelled /dev, /dev/shm, /run, /sys/fs/cgroup in 25.046ms.
[    8.987389] systemd[1]: systemd v250.8-1.fc36 running in system mode (...)
```

So, relabeling would be needed only for other mount points. And in fact, I would expect systemd to take care of that too (`/rw` and `/home` could still require relabeling, as those may be mounted with `mount` command directly and not by systemd).